### DISCUSSION TITLE
Value formatter per Item position not the item value

### DISCUSSION THREAD

**Discussion by ma7moudovic** [patrykandpatrick/vico#435] [Questions] [answered]
In X axis how could I show a formatted value per index not per the value
for ex: I have this dataset
`[
  {
    "date": "Jan",
    "revenue": 10
  },
  {
    "date": "Feb",
    "revenue": 40
  },
  {
    "date": "Mar",
    "revenue": 30
  },
  {
    "date": "Apr",
    "revenue": 20
  }
]`

and I want to show on the X axis [Jan] [Feb] [Mar] [Apr]
and on Y axis [10] [20] [30] [40]

**Comment by patrickmichalik** [comment]
Hello! Here’s a simple example. You’ve mentioned that the vertical axis should show labels for 10, 20, 30, and 40. Here, zero is labeled too. If you’d like to remove this label, a custom `AxisItemPlacer.Vertical` implementation will be required.

```kt
data class RevenueInfo(val date: String, val revenue: Float)

val data = listOf(RevenueInfo("Jan", 10f), RevenueInfo("Feb", 40f), RevenueInfo("Mar", 30f), RevenueInfo("Apr", 20f))
val chartEntryModel = entryModelOf(data.mapIndexed { index, revenueInfo -> entryOf(index, revenueInfo.revenue) })
val bottomAxisValueFormatter = AxisValueFormatter<AxisPosition.Horizontal.Bottom> { x, _ -> data[x.toInt()].date }
val startAxisItemPlacer = AxisItemPlacer.Vertical.default(maxItemCount = 5)

@Composable
@Preview(showBackground = true, backgroundColor = 0xffffffff)
fun Example() {
    Chart(
        chart = columnChart(),
        model = chartEntryModel,
        startAxis = rememberStartAxis(itemPlacer = startAxisItemPlacer),
        bottomAxis = rememberBottomAxis(valueFormatter = bottomAxisValueFormatter),
    )
}
```

<img src="https://github.com/patrykandpatrick/vico/assets/120058021/3b64bf14-9ab1-408b-b1b5-bff3c1bdc607" width="720" />

**Comment by ma7moudovic** [comment]
@patrickmichalik thanks a lot for your fast response, really appreciated ♥️
what if I want the chart to be StackedChart and my data set gonna be like this
 val data = listOf(
                RevenueInfo(date = "Jan", revenue = 10f, expectedRevenue = 5f),
                RevenueInfo(date = "Feb", revenue = 30f, expectedRevenue = 15f),
                RevenueInfo(date = "Mar", revenue = 20f, expectedRevenue = 10f),
                RevenueInfo(date = "Apr", revenue = 50f, expectedRevenue = 20f)
        )

**Comment by ma7moudovic** [comment, marked as answer]
val chartEntryModel = entryModelOf(data.mapIndexed { index, revenueInfo ->
            entryOf(index, revenueInfo.revenue)
        }, data.mapIndexed { index, revenueInfo ->
            entryOf(index, revenueInfo.alternativeRevenue)
        })
        
@patrickmichalik this one worked for me 🎉 
thanks a lot

**Reply by patrickmichalik** [reply]
Yup, looks good. You’re welcome!
